spec/tools/miqldap_to_sssd/miqldap_configuration_spec.rb

```diff
@@ -4,19 +4,45 @@
 
 describe MiqLdapToSssd::MiqLdapConfiguration do
   describe '#retrieve_initial_settings' do
+    let(:settings) { {:tls_cacert => 'cert', :basedn_domain => "example.com"} }
+
     it 'raises an error when the basedn domain can not be determined' do
       expect(MiqLdapToSssd::LOGGER).to receive(:fatal)
-      subject = described_class.new(:basedn => nil, :basedn_domain => nil)
+      subject = described_class.new(settings.merge(:basedn => nil, :basedn_domain => nil))
       expect { subject.retrieve_initial_settings }.to raise_error(MiqLdapToSssd::MiqLdapConfigurationArgumentError)
     end
 
-    it 'does not modify basedn_domain if providedn' do
-      subject = described_class.new(:basedn_domain => "example.com")
+    it 'when mode is ldap and bind dn is nil raises an error' do
+      expect(MiqLdapToSssd::LOGGER).to receive(:fatal)
```

Comment:
Is this important to your test?

Reply:
@bdunne Thank you for the question.

Yes, the bind dn must be provided. It can be provided in 1 of 2 ways.

1 - From the appliance configuration as stored in: Settings.authentication[:bind_dn]

2- If not configured on the appliance the user must provide it as an argument to the  miqldap_to_sssd script.

If it is not provided in either of these two ways the miqldap_to_sssd script will raise an error indicating it is missing.